python-frontend/src/main/java/org/sonar/python/types/TypeShedVisitor.java

```diff
@@ -0,0 +1,96 @@
+
+import com.sonar.sslr.api.AstNode;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.python.api.tree.ClassDef;
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.plugins.python.api.tree.FunctionDef;
+import org.sonar.plugins.python.api.tree.TypeAnnotation;
+import org.sonar.python.parser.PythonParser;
+import org.sonar.python.semantic.FunctionSymbolImpl;
+import org.sonar.python.semantic.SymbolImpl;
+import org.sonar.python.tree.PythonTreeMaker;
+
+import static org.sonar.plugins.python.api.symbols.Symbol.Kind.OTHER;
+
+public class TypeShedVisitor extends BaseTreeVisitor {
+
+  private Map<String, Symbol> symbolsByName = new HashMap<>();
+  private static final Logger LOG = Loggers.get(TypeShedVisitor.class);
+  private static Map<String, Symbol> typeShedSymbols;
+
+  private TypeShedVisitor() {
+  }
+
+  public static Map<String, Symbol> typeShedSymbols() {
+    if (TypeShedVisitor.typeShedSymbols == null) {
+      Map<String, Symbol> typeShedSymbols = Collections.emptyMap();
+      InputStream resource = TypeShedVisitor.class.getResourceAsStream("builtins.pyi");
+      try (BufferedReader reader = new BufferedReader(new InputStreamReader(resource, StandardCharsets.UTF_8))) {
+        String builtins = reader.lines().collect(Collectors.joining("\n"));
+        AstNode astNode = PythonParser.create().parse(builtins);
+        FileInput fileInput = new PythonTreeMaker().fileInput(astNode);
+        TypeShedVisitor typeShedVisitor = new TypeShedVisitor();
+        fileInput.accept(typeShedVisitor);
+        typeShedSymbols = typeShedVisitor.symbolsByName;
+      } catch (IOException | UncheckedIOException e) {
+        LOG.debug("Unable to read builtin types.");
+      }
+      TypeShedVisitor.typeShedSymbols = Collections.unmodifiableMap(typeShedSymbols);
+    }
+    return typeShedSymbols;
+  }
+
+  @Override
+  public void visitFunctionDef(FunctionDef functionDef) {
+    String functionName = functionDef.name().name();
+    SymbolImpl symbol = (SymbolImpl) symbolsByName.get(functionName);
+    if (symbol != null) {
+      symbol.setKind(OTHER);
+    } else {
+      FunctionSymbolImpl functionSymbol = new FunctionSymbolImpl(functionName, functionName, false, false, false, Collections.emptyList(), null);
+      TypeAnnotation returnTypeAnnotation = functionDef.returnTypeAnnotation();
+      if (returnTypeAnnotation != null) {
+        functionSymbol.setInferredReturnType(InferredTypes.declaredType(returnTypeAnnotation));
```

Comment:
Is `setInferredReturnType` the right name? Shouldn't it be `setDeclaredReturnType`? We could imagine that some day we compare the return type which is declared in a function declaration with the return type which we can infer from its body.
```
def f() -> str:
  return 42
```